src/utils/crypto.ts

```diff
@@ -0,0 +1,140 @@
+export type JwkKeyPair = {
+    publicKey: JsonWebKey;
+    privateKey: JsonWebKey;
+};
+
+export interface JsonWebKeyPairStorage {
+    loadKeyPair(): Promise<JwkKeyPair | undefined>;
+    saveKeyPair(keyPair: JwkKeyPair): Promise<void>;
+}
+
+function arraybuffer2hex(ab: ArrayBufferLike) {
+    return Array.prototype.map
+        .call(new Uint8Array(ab), x =>
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
+            ('00' + x.toString(16)).slice(-2),
+        )
+        .join('');
+}
+
+export class ConnectCrypto {
+    #subtleCrypto: Promise<SubtleCrypto>;
+    #keyPair: CryptoKeyPair | undefined;
+    #keyPairStorage: JsonWebKeyPairStorage;
+
+    static algorithm: EcKeyAlgorithm = {
+        name: 'ECDSA',
+        namedCurve: 'P-256',
+    };
+
+    constructor(keyPairStorage: JsonWebKeyPairStorage) {
+        this.#keyPairStorage = keyPairStorage;
+
+        // Initializes the subtleCrypto module according to the host environment
+        this.#subtleCrypto = this.#initCrypto();
+    }
+
+    #initCrypto() {
+        return new Promise<SubtleCrypto>((resolve, reject) => {
+            if (typeof window === 'undefined') {
+                import('node:crypto')
+                    .then(module => {
+                        resolve(module.webcrypto.subtle as SubtleCrypto);
+                    })
+                    .catch(reject);
+            } else {
+                // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+                if (!window.crypto?.subtle) {
+                    reject(new Error('Crypto module not available'));
+                }
+                resolve(window.crypto.subtle);
+            }
+        });
+    }
+
+    // loads the key pair from storage or generates a new one if none is stored
+    async initialize() {
+        const loadedKeyPair = await this.#keyPairStorage.loadKeyPair();
+        if (loadedKeyPair) {
+            this.#keyPair = await this.#importKeyPair(loadedKeyPair);
+            console.log('Loaded key pair', await this.publicKey());
+        } else {
+            this.#keyPair = await this.#generateECDSAKeyPair();
+            console.log('Created key pair', await this.publicKey());
+            await this.#keyPairStorage.saveKeyPair(await this.#exportKeyPair());
+        }
+    }
+
+    async publicKey() {
+        if (!this.#keyPair) {
+            return undefined;
+        }
+
+        const subtleCrypto = await this.#subtleCrypto;
+        const publicKey = await subtleCrypto.exportKey(
+            'raw',
+            this.#keyPair.publicKey,
+        );
+        return arraybuffer2hex(publicKey);
```

Comment:
This returns a 130 character public key.
I believe it is possible to compress it or even to do something similar to what Ethereum does to create a 42 char address from a public key.
Need to understand how it works in the signature verification process.